starter_code/src/movies.js

```diff
@@ -1,19 +1,110 @@
 /* eslint no-restricted-globals: 'off' */
+// let movies = [
+//     {
+//       title: 'The Shawshank Redemption',
+//       year: '1994',
+//       director: 'Frank Darabont',
+//       duration: '2h 22min',
+//       genre: ['Crime', 'Drama'],
+//       rate: '9.3'
+//     },
 
 // Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+function ratesAverage(movies) {
+    // check if array is not empty
+    if (!movies.length) return 0;
+
+    // create empty array
+    let movieRates = [];
+
+    // push all rates into array
+    movies.forEach(function (movie) {
```

Comment:
In this case, a map would have achieved it much quicker.
.map(movie => movie.rate